src/tic-tac-toe.ts

```diff
@@ -40,35 +43,73 @@ class Game {
     return square === this.currentPlayer;
   };
 
-  private checkRowsForWin(): boolean {
-    return this.board.some(row => row.every(this.isCurrentPlayer));
+  private checkRowsForWin(): WinningVector | undefined {
+    for (let rowIndex = 0; rowIndex < this.board.length; rowIndex++) {
+      const row = this.board[rowIndex];
+      const vector: WinningVector = [];
+      for (let colIndex = 0; colIndex < this.board.length; colIndex++) {
+        const square = row[colIndex];
+        if (this.isCurrentPlayer(square)) {
+          vector.push({ row: rowIndex, col: colIndex });
+        }
+      }
+      if (vector.length === this.boardSize) {
+        return vector;
+      }
+    }
+    return;
   }
 
-  private checkColumnsForWin(): boolean {
+  private checkColumnsForWin(): WinningVector | undefined {
     const transposedBoard = this.eachIndex().map(i =>
       this.board.map(row => row[i])
     );
 
-    return transposedBoard.some(row => row.every(this.isCurrentPlayer));
+    for (let colIndex = 0; colIndex < transposedBoard.length; colIndex++) {
+      const column = transposedBoard[colIndex];
+      const vector: WinningVector = [];
+
+      for (let rowIndex = 0; rowIndex < transposedBoard.length; rowIndex++) {
+        const square = column[rowIndex];
+        if (this.isCurrentPlayer(square)) {
+          vector.push({ row: rowIndex, col: colIndex });
+        }
+      }
+      if (vector.length === this.boardSize) {
+        return vector;
+      }
+    }
+    return;
   }
 
-  private checkDiagonalsForWin(): boolean {
-    const diagonals = [
-      this.eachIndex().map(i => this.board[i][i]),
-      this.eachIndex().map(i => this.board[i][this.boardSize - 1 - i]),
-    ];
+  private checkDiagonalsForWin(): WinningVector | undefined {
+    const winningVector: WinningVector = [];
 
-    return diagonals.some(values => values.every(this.isCurrentPlayer));
+    for (let i = 0; i < this.boardSize; i++) {
+      // check values from top left to bottom right (ie for 3x3 grid => [0, 0], [1, 1], [2, 2])
+      if (this.isCurrentPlayer(this.board[i][i])) {
+        winningVector.push({ row: i, col: i });
+        // check values from top right to bottom left (ie for 3x3 grid => [0, 2], [1, 1], [2, 0])
+      } else if (this.isCurrentPlayer(this.board[i][this.boardSize - 1 - i])) {
```

Comment:
It feels like there's a bug here due to the fact that your win condition is simply the vector length, but you're pushing into it from either the L-R diagonal or the R-L diagonal at the same time. Consider this board:

```
x o o
_ x _
x _ _ 
```
I believe that this would count as a "win" due to the way your loop works. If so, I would just slightly modify the original algorithm:

```jsx
const Diagonals: WinningVector[] = [
  this.eachIndex().map(i => [i, i]),
  this.eachIndex().map(i => [i, this.boardSize - i - 1]),
];

return diagonals.find(diagonal => diagonal.every(([row, col]) => this.isCurrentPlayer(this.board[row][col]));
```

Reply:
This seems to fix that bug with the false positive. I just had to modify it to use the new way that the winVector is stored => `{row: 0, col: 0}`